<?xml version="1.0" encoding="UTF-8"?>
<launch>
    <!-- arguments -->
    <!-- Jang2009 -->
    <arg name="gamma_1" default="0.3" />
    <arg name="gamma_2" default="3.0" />
    <arg name="h" default="1.0" />
    <!-- Kim2002_1 and Kim2002_2 -->
    <arg name="k" default="0.5" />
    <arg name="mu" default="1.0" />

    <arg name="gui" default="true" />
    <arg name="world_name" default="$(find gazebo_ros)/worlds/empty.world" />
    <arg name="front_laser" default="false" />

    <!-- spawn Jackal robot -->
    <include file="$(find jackal_gazebo)/launch/empty_world.launch">
        <arg name="gui" value="$(arg gui)" />
        <arg name="world_name" value="$(arg world_name)" />
        <arg name="front_laser" value="$(arg front_laser)" />
    </include>

    <!-- run controller -->
    <node pkg="ss_gazebo" type="wheeled_mobile_robot_pose_control_node"
            name="wheeled_mobile_robot_pose_control_node" respawn="false" output="screen">
        <param name="wheeled_mobile_robot_pose_control/Jang2009/gamma_1" type="double" value="$(arg gamma_1)" />
        <param name="wheeled_mobile_robot_pose_control/Jang2009/gamma_2" type="double" value="$(arg gamma_2)" />
        <param name="wheeled_mobile_robot_pose_control/Jang2009/h" type="double" value="$(arg h)" />
        <param name="wheeled_mobile_robot_pose_control/Kim2002_1/k" type="double" value="$(arg k)" />
        <param name="wheeled_mobile_robot_pose_control/Kim2002_1/mu" type="double" value="$(arg mu)" />
        <param name="wheeled_mobile_robot_pose_control/Kim2002_2/k" type="double" value="$(arg k)" />
        <param name="wheeled_mobile_robot_pose_control/Kim2002_2/mu" type="double" value="$(arg mu)" />
    </node>
</launch>